src/java.base/share/native/libjli/java.c

```diff
@@ -72,6 +77,49 @@
 
 #define USE_STDERR JNI_TRUE     /* we usually print to stderr */
 #define USE_STDOUT JNI_FALSE
+#if defined(J9VM_OPT_CRAC_SUPPORT)
+/* The return codes indicating errors, successful execution, or specific conditions. */
+#define EXECVP_ERROR -1
+#define EXECVP_SUCCESS 0
+#define WAIT_INDEFINITELY 0
+#define MEMORY_ALLOCATION_ERROR 1
+#define OPTION_NAME_NOT_FOUND_ERROR 2
+#define OPTION_VALUE_NOT_FOUND_ERROR 3
+#define OPTION_VALUE_NOT_VALID_ERROR 4
+#define OPTION_VALUE_NOT_EXPECTED_ERROR 5
+#define COMMAND_OPTION_LENGTH_CALCULATION_ERROR 6
+
+/* The error messages. */
+#define OPTION_VALUE_NOT_FOUND_ERROR_MESSAGE "The value of the command line option %s was not found."
+#define OPTION_VALUE_NOT_VALID_ERROR_MESSAGE "The value of the command line option is not valid, option=%s, value=%s."
+#define OPTION_VALUE_NOT_EXPECTED_ERROR_MESSAGE "The value of the command line option is not expected, option=%s, value=%s."
+#define CHECKPOINT_DIRECTORY_ERROR_MESSAGE "Failed to get the CRIU checkpoint directory, error=%d."
+#define LOG_LEVEL_ERROR_MESSAGE "Failed to get the CRIU log level, error=%d."
+#define UNPRIVILEGED_MODE_ERROR_MESSAGE "Failed to get the CRIU unprivileged mode, error=%d."
+#define LOG_FILE_ERROR_MESSAGE "Failed to get the CRIU log file, error=%d."
+#define COMMAND_OPTION_LENGTH_CALCULATION_ERROR_MESSAGE "Failed to calculate the length of the command option, value=%s, format=%s."
+#define COMMAND_OPTION_MEMORY_ALLOCATION_ERROR_MESSAGE "Failed to allocate memory for the command option, value=%s, format=%s."
+#define RESTORE_FROM_CHECKPOINT_ERROR_MESSAGE "Failed to restore from checkpoint, error=%d."
+#define RESTORE_CHILD_PROCESS_FAILED_ERROR_MESSAGE "The CRIU restore child process failed."
+
+/* The option names. */
+#define CRAC_RESTORE_FROM_OPTION_NAME "-XX:CRaCRestoreFrom"
+#define LOG_LEVEL_OPTION_NAME "-Dopenj9.internal.criu.logLevel"
+#define UNPRIVILEGED_MODE_OPTION_NAME "-Dopenj9.internal.criu.unprivilegedMode"
+#define LOG_FILE_OPTION_NAME "-Dopenj9.internal.criu.logFile"
+
+/* The option formats. */
+#define LOG_LEVEL_OPTION_FORMAT "-v%d"
+#define LOG_FILE_OPTION_FORMAT "--log-file=%s"
+
+/* The CRIU command options. */
+#define CRIU_COMMAND "criu"
+#define CRIU_RESTORE_OPTION "restore"
+#define CRIU_CHECKPOINT_DIRECTORY_OPTION "-D"
+#define CRIU_DEFAULT_LOG_LEVEL_OPTION "-v2"
+#define CRIU_SHELL_JOB_OPTION "--shell-job"
+#define CRIU_UNPRIVILEGED_MODE_OPTION "--unprivileged"
+#endif /* defined(J9VM_OPT_CRAC_SUPPORT) */
```

Comment:
These are (or will be) unused and should simply be removed:
* `COMMAND_OPTION_LENGTH_CALCULATION_ERROR`
* `COMMAND_OPTION_MEMORY_ALLOCATION_ERROR_MESSAGE`
* `LOG_FILE_OPTION_FORMAT`
* `LOG_LEVEL_OPTION_FORMAT`
* `MEMORY_ALLOCATION_ERROR`

Many others are used only once: I suggest it's clearer, for example, to just write `"--unprivileged"` rather than hide that text in `CRIU_UNPRIVILEGED_MODE_OPTION`, define a long from its use.

Reply:
How can I request clarification as it seems like the suggestion is to remove all the macros by hardcoding the strings and I do not understand the following part of the [comment](https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/780#discussion_r1638816318):

```
 ..., define a long from its use.
```

?

Reply:
@tajila @keithc-ca how can I know the answer to the following questions: how can I request clarification as it seems like the suggestion is to remove all the macros by hardcoding the strings and I do not understand the following part of the [comment](https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/780#discussion_r1638816318):

```
 ..., define a long from its use.
```

?

Reply:
It seems like I have addressed the comment.